htlcswitch/mailbox.go

```diff
@@ -710,9 +715,20 @@ func (m *memoryMailBox) FailAdd(pkt *htlcPacket) {
 	// peer if this is a forward, or report to the user if the failed
 	// payment was locally initiated.
 	var failure lnwire.FailureMessage
-	update, err := m.cfg.fetchUpdate(m.cfg.shortChanID)
-	if err != nil {
-		failure = &lnwire.FailTemporaryNodeFailure{}
+
+	// Try to use the failAliasUpdate function in case this is an
+	// option-scid-alias channel. If it's not, we'll fallback to the
+	// original non-alias behavior.
+	update := m.cfg.failAliasUpdate(pkt.originalOutgoingChanID, false)
```

Comment:
Similar question here re pushing this logic deeper down into the call stack, so we can reduce the clutter at this abstraction level and hide more details.

Reply:
will add to TODO, can be done. mailbox doesn't even use the `incoming` boolean in the func since it's only outgoing

Reply:
done